### DISCUSSION TITLE
分p视频怎么下载

### DISCUSSION THREAD

**Discussion by swttkonjjj** [yutto-dev/yutto#129] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/66727004/235571616-9d2ca8f6-d927-4df6-9980-5dfdd25c2ba2.png)
这种分p视频只会下载p1的视频

**Comment by SigureMo** [comment, marked as answer]
`yutto` 默认是只下载单 p 的，比如 `yutto https://www.bilibili.com/video/BV1vZ4y1M7mQ` 只会下载 BV1vZ4y1M7mQ 的第一 p，可以通过调整 url 来下载不同 p `yutto https://www.bilibili.com/video/BV1vZ4y1M7mQ?p=2` 即是第二 p。

`yutto` 还提供了 `-b` 参数来下载多 P，比如 `yutto -b https://www.bilibili.com/video/BV1vZ4y1M7mQ` 则是解析该 url 对应的播放列表，进行批量下载，对于多 p 视频来说，就是同时将所有 P 都下载下来，在这种情况下，你可以通过 `-p` 参数来控制想要下载第几 p，比如 `yutto -b https://www.bilibili.com/video/BV1vZ4y1M7mQ -p 10` 即是下载第 10 P

**Reply by swttkonjjj** [reply]
``yutto https://www.bilibili.com/video/BV1vZ4y1M7mQ?p=2`` 使用这种格式还是会下载第一p

**Reply by SigureMo** [reply]
看样子是 B 站修改了 url，现在会多一个 `/`，变成 `https://www.bilibili.com/video/BV1vZ4y1M7mQ/?p=2`，问题已经在 59e2d524ffd168618203d1be3c0ee1e84ea6cc41 修复

直接用批量下载吧，批量下载应该是没问题的
